### PR TITLE
Added loading to dropdowns

### PR DESCRIPTION
**Pull request by aasimkhan30**
Screenshots: 
Dark:
Loading state:
image
Regular state:
image
Light:
Loading state:
image
Regular state:
image

**Example:** 
image

### CODE REVIEW

**Review by alanrenmsft — CHANGES_REQUESTED** [review]
I guess it is ok to push the loading logic to the dropdown component, but please refer to the loading component for a complete set of properties to be fully accessible. also please don't use inputbox as the loading indicator, internally you can still wrap the dropdown component with the loading component.

**Comment by aaomidi** [comment]
I like this but I'd defer to what @alanrenmsft is bringing up about accessibility

File: src/sql/workbench/browser/modelComponents/dropdown.component.ts [hunk 1]
```diff
@@ -103,6 +110,12 @@ export default class DropDownComponent extends ComponentBase<azdata.DropDownProp
 			}));
 			this._validations.push(() => !this.required || this.editable || !!this._selectBox.value);
 		}
+
+		this._loadingBox = new SelectBox([this.getStatusText()], this.getStatusText(), this.contextViewService, this._loadingBoxContainer.nativeElement);
+		this._loadingBox.render(this._loadingBoxContainer.nativeElement);
+		this._register(this._loadingBox);
+		this._register(attachSelectBoxStyler(this._loadingBox, this.themeService));
+		this._loadingBoxContainer.nativeElement.className = ''; // Removing the dropdown arrow icon from the right
```
Comment (aaomidi) [inline src/sql/workbench/browser/modelComponents/dropdown.component.ts:118]: Could you instead potentially add a new class that removes the dropdown arrow icon in CSS? NBD if not

File: src/sql/workbench/browser/modelComponents/dropdown.component.ts [hunk 2]
```diff
@@ -198,11 +220,15 @@ export default class DropDownComponent extends ComponentBase<azdata.DropDownProp
...
+		return (this.editable && !this._isInAccessibilityMode) && !this.loading ? '' : 'none';
 	}
 
 	public getNotEditableDisplay(): string {
-		return !this.editable || this._isInAccessibilityMode ? '' : 'none';
+		return (!this.editable || this._isInAccessibilityMode) && !this.loading ? '' : 'none';
+	}
+
+	public getLoadingDisplay(): string {
+		return this.loading ? '' : 'none';
```
Comment (aaomidi) [inline src/sql/workbench/browser/modelComponents/dropdown.component.ts:231]: localized strings?

Comment (aasimkhan30) [reply src/sql/workbench/browser/modelComponents/dropdown.component.ts:231]: That is the css display Property

**Review by aaomidi — APPROVED** [review]
Just to clarify, this doesn't change the behavior of existing dropdowns, right? Only if you want to use loading features this comes in?

File: src/sql/workbench/browser/modelComponents/dropdown.component.ts (hunk 2 above)
Comment (aaomidi) [reply src/sql/workbench/browser/modelComponents/dropdown.component.ts:231]: Makes sense, thanks for clarifying!

**Comment by aasimkhan30** [comment]
> Just to clarify, this doesn't change the behavior of existing dropdowns, right? Only if you want to use loading features this comes in?

@aaomidi Nope. it should not change any existing behavior. It just adds a new loading dropdown that will be visible when loading = true.

**Pull request merged**